Properly set modified for variety of cadence passes

DrJessop opened the pull request:
Summary: Update a large batch of passes to use the new remove or replace interface

Differential Revision: D85817305

abeakkas reviewed (approved):
With pending Phabricator comment.

DrJessop committed:
Use new RemoveOrReplace interface for a variety of remove passes (#15471)

Summary:

Update a large batch of passes to use the new remove or replace interface

Reviewed By: abeakkas

Differential Revision: D85817305
